packages/cli/src/dbt/targets/Bigquery/index.ts

```diff
@@ -79,43 +82,58 @@ export const convertBigquerySchema = async (
     target: Target,
 ): Promise<CreateBigqueryCredentials> => {
     const validate = ajv.compile<BigqueryTarget>(bigqueryTargetJsonSchema);
+    let getKeyfileContents : (target: Target) => Promise<Record<string, string>> | Promise<OAuth2ClientOptions>;
     if (validate(target)) {
-        let getBigqueryCredentials;
         switch (target.method) {
             case 'oauth':
-                getBigqueryCredentials = getBigqueryCredentialsFromOauth;
+                getKeyfileContents = getBigqueryCredentialsFromOauth;
                 break;
             case 'service-account':
-                getBigqueryCredentials =
+                getKeyfileContents =
                     getBigqueryCredentialsFromServiceAccount;
                 break;
             case 'service-account-json':
-                getBigqueryCredentials =
+                getKeyfileContents =
                     getBigqueryCredentialsFromServiceAccountJson;
                 break;
+            case 'oauth-secrets':
+                getKeyfileContents = getBigqueryCredentialsFromOauthSecrets;
+                break;
             default:
                 throw new ParseError(
                     `BigQuery method ${target.method} is not yet supported`,
                 );
         }
 
-        if (target.project === undefined && target.method !== 'oauth')
+        if (target.project === undefined && target.method !== 'oauth') {
             throw new ParseError(
                 `BigQuery project is required for ${target.method} authentication method`,
             );
+        }
 
-        return {
+        const creds: CreateBigqueryCredentials = {
             type: WarehouseTypes.BIGQUERY,
             project: target.project || '',
             dataset: target.dataset || target.schema,
             timeoutSeconds: target.timeout_seconds,
             priority: target.priority,
-            keyfileContents: await getBigqueryCredentials(target),
             retries: target.retries,
             maximumBytesBilled: target.maximum_bytes_billed,
             location: target.location,
             executionProject: target.execution_project,
         };
+
+        if (getKeyfileContents) {
+            const keyfile = await getKeyfileContents(target);
+
+            if ('credentials' in keyfile) {
+                creds.authClient = new OAuth2Client(keyfile);
```

Comment:
Could we store the OAuth2ClientOptions instead of the OAuth2Client instance? 
I'm surprised saving the instanced worked when creating projects/previews as the credentials are sent to the server/database, to later be used in the UI.